src/deepsparse/server/main.py

```diff
@@ -214,7 +214,15 @@ def server_app_factory():
     _LOGGER.debug("loaded server config %s", config)
     _add_general_routes(app, config)
 
-    pipelines = [Pipeline.from_config(model_config) for model_config in config.models]
+    num_cores = None
+    for model_config in config.models:
+        num_cores = num_cores or model_config.num_cores
```

Comment:
it'd be nice to expand the Context to run on the largest num_cores specified
```suggestion
    num_cores = None
    for model_config in config.models:
        num_cores = max(num_cores, model_config.num_cores)
```